tsdb/db_test.go

```diff
@@ -682,6 +682,32 @@ func TestDB_Snapshot(t *testing.T) {
 	require.Equal(t, 1000.0, sum)
 }
 
+func TestDB_BeyondTimeRetention(t *testing.T) {
+	opts := DefaultOptions()
+	opts.RetentionDuration = 100
+	db := openTestDB(t, opts, nil)
+	defer func() {
+		require.NoError(t, db.Close())
+	}()
+
+	metas := []BlockMeta{
```

Comment:
I would like a little explanation what is going on here - why these 4 blocks, why in this order, what are we expecting.

Reply:
added some notes